tests/functional/generate_pages.py

```diff
@@ -409,6 +427,13 @@ def process_final_summary(schema_data, page_spec, collapsible, section_summary=F
         page_spec.write(COLLAPSIBLE_SUMMARY_GETTER)
 
 
+def process_definition(index, page_spec):
+    definition_context = {"definitionIndex": index}
+    page_spec.write(DEFINITION_TITLE_GETTER.substitute(definition_context))
+    page_spec.write(DEFINITION_CONTENT_GETTER.substitute(definition_context))
+    page_spec.write(DEFINITION_BUTTON_GETTER.substitute(definition_context))
```

Comment:
The existing definition uses an index to get the data attributes like this:
```js
definitionTitle(definitionIndex) { return `[data-qa='question-definition-0-title']`; }
```
However, as you can see the existing one is wrong because it should actually be:
```js
definitionTitle(definitionIndex) { return `[data-qa='question-definition-${definitionIndex}-title']`; }
```
I think we should update it to do the same, so we don't have 2 ways of processing a definition. You could use the existing `Template` partial for both question definition and definition in content pages. The only thing you have to do is pass in the "id" where id is either "definition" or "question-definition". 

I have made this git gist to demo what I mean. It fixes the existing definition while make both definitions types reuse the same template.
https://gist.github.com/MebinAbraham/6449e983bb01d09f05cd8757b5acdaa7